crates/router/src/core/routing.rs

```diff
@@ -84,12 +84,13 @@ impl RoutingAlgorithmUpdate {
 pub async fn retrieve_merchant_routing_dictionary(
     state: SessionState,
     merchant_account: domain::MerchantAccount,
+    authentication_profile_id: Option<common_utils::id_type::ProfileId>,
```

Comment:
This should accept a vector of profile_ids.
Ref: https://github.com/juspay/hyperswitch/blob/379831932a9c1e73b669f40af75cd02e682a0ec7/crates/router/src/core/payments.rs#L2917